agent/api/ecsclient/client.go

```diff
@@ -212,11 +261,23 @@ func getCpuAndMemory() (int64, int64) {
 	return int64(cpu), mem
 }
 
-func getAdditionalAttributes() []*ecs.Attribute {
-	return []*ecs.Attribute{&ecs.Attribute{
+func (client *APIECSClient) getAdditionalAttributes() []*ecs.Attribute {
+	attributes := []*ecs.Attribute{&ecs.Attribute{
```

Comment:
You're already doing this elsewhere, but you should be able to get rid of the duplicate type declaration.

```go
attributes := []*ecs.Attribute{{
	// ...
}}
```